Sources/XMTPiOS/Conversations.swift

```diff
@@ -99,33 +99,33 @@ public actor Conversations {
 	public func streamAllMessages() async throws -> AsyncThrowingStream<DecodedMessage, Error> {
 		return AsyncThrowingStream { continuation in
 			Task {
-				while true {
-					var topics: [String] = [
-						Topic.userInvite(client.address).description,
-						Topic.userIntro(client.address).description,
-					]
+				var topics: [String] = [
+					Topic.userInvite(client.address).description,
+					Topic.userIntro(client.address).description,
+				]
 
-					for conversation in try await list() {
-						topics.append(conversation.topic)
-					}
+				for conversation in try await list() {
+					topics.append(conversation.topic)
+				}
 
+				while(true) {
 					do {
-						for try await envelope in client.subscribe(topics: topics) {
+						for try await (envelope, subscription) in client.subscribe(topics: topics) {
 							if let conversation = conversationsByTopic[envelope.contentTopic] {
 								let decoded = try conversation.decode(envelope)
 								continuation.yield(decoded)
 							} else if envelope.contentTopic.hasPrefix("/xmtp/0/invite-") {
 								let conversation = try fromInvite(envelope: envelope)
 								conversationsByTopic[conversation.topic] = conversation
-								break // Break so we can resubscribe with the new conversation
+								topics.append(conversation.topic)
+								try await subscription.update(req: client.makeSubscribeRequest(topics: topics).toFFI)
```

Comment:
Does this update method want the entire list of topics or just the new topics?

Reply:
This method is also blocking from the for loop continuing..

Reply:
It takes the entire list of topics